airflow-core/src/airflow/ui/src/components/DateTimeInput.tsx

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Input, type InputProps } from "@chakra-ui/react";
+import dayjs from "dayjs";
+import tz from "dayjs/plugin/timezone";
+import { forwardRef } from "react";
+
+import { useTimezone } from "src/context/timezone";
+
+dayjs.extend(tz);
+
+type Props = {
+  readonly value: string;
+} & InputProps;
+
+export const DateTimeInput = forwardRef<HTMLInputElement, Props>(({ onChange, value, ...rest }, ref) => {
+  const { selectedTimezone } = useTimezone();
+
+  // Make the value timezone-aware
+  const date = dayjs(value).tz(selectedTimezone).format("YYYY-MM-DDTHH:mm:ss.SSS");
+
+  return (
+    <Input
```

Comment:
For this new type, can you add some details as type into the params docs as well?
airflow-core/docs/core-concepts/params.rst

And would be cool to add an example to airflow-core/src/airflow/example_dags/example_params_ui_tutorial.py as well.

Reply:
Nothing is new. We just make datetime inputs aware of the user defined timezone

Reply:
For me this is new. So far in the params docs it is described that you can use "date", "time", and "date-time". "datetime-local" is new (as form element), So as long as no example or no docs, probably nobody will know about this trigger form element feature.